core/logger/test_logger.go

```diff
@@ -67,11 +68,11 @@ func MemoryLogTestingOnly() *MemorySink {
 // TestLogger creates a logger that directs output to PrettyConsole configured
 // for test output, and to the buffer testMemoryLog. t is optional.
 // Log level is derived from the LOG_LEVEL env var.
-func TestLogger(t T) Logger {
+func TestLogger(t T, cores ...zapcore.Core) Logger {
```

Comment:
What about something like this to keep the API simple?
```suggestion
func TestLogger(t T) Logger { return testLogger(t) }
		
func TestLoggerObserved(t T, lvl zapcore.Level) (Logger, *observer.ObservedLogs) {
	observedZapCore, observedLogs := observer.New(lvl)
	return testLogger(t, observedZapCore), observedLogs
}

func testLogger(t T, cores ...zapcore.Core) Logger {
```